crates/forge/tests/e2e/clean.rs

```diff
@@ -0,0 +1,241 @@
+    };
+
+    assert_eq!(
+        check_clean_components_state(temp_dir.path()),
+        clean_components_state
+    );
+}
+
+#[test]
+#[cfg(not(target_os = "windows"))]
+fn test_clean_profile() {
+    let temp_dir = setup_package("coverage_project");
+
+    let clean_components_state = CleanComponentsState {
+        coverage: false,
+        profile: true,
+        cache: true,
+        trace: true,
+    };
+
+    generate_clean_components(clean_components_state, &temp_dir);
+
+    runner(&temp_dir)
+        .arg("clean")
+        .arg("profile")
+        .assert()
+        .success();
+
+    let clean_components_state: CleanComponentsState = CleanComponentsState {
+        coverage: false,
+        profile: false,
+        cache: true,
+        trace: true,
+    };
+
+    assert_eq!(
+        check_clean_components_state(temp_dir.path()),
+        clean_components_state
+    );
+}
+
+#[test]
+fn test_clean_cache() {
+    let temp_dir = setup_package("coverage_project");
+
+    let clean_components_state = CleanComponentsState {
+        coverage: false,
+        profile: false,
+        cache: true,
+        trace: false,
+    };
+
+    generate_clean_components(clean_components_state, &temp_dir);
+
+    runner(&temp_dir)
+        .arg("clean")
+        .arg("cache")
+        .assert()
+        .success();
+
+    let clean_components_state: CleanComponentsState = CleanComponentsState {
+        coverage: false,
+        profile: false,
+        cache: false,
+        trace: false,
+    };
+
+    assert_eq!(
+        check_clean_components_state(temp_dir.path()),
+        clean_components_state
+    );
+}
+
+#[test]
+#[cfg_attr(not(feature = "scarb_2_8_3"), ignore)]
+fn test_clean_all() {
+    let temp_dir = setup_package("coverage_project");
+
+    let clean_components_state = CleanComponentsState {
+        coverage: true,
+        cache: true,
+        trace: true,
+        profile: true,
+    };
+
+    generate_clean_components(clean_components_state, &temp_dir);
+
+    runner(&temp_dir).arg("clean").arg("all").assert().success();
+
+    let clean_components_state: CleanComponentsState = CleanComponentsState {
+        coverage: false,
+        cache: false,
+        trace: false,
+        profile: false,
+    };
+
+    assert_eq!(
+        check_clean_components_state(temp_dir.path()),
+        clean_components_state
+    );
+}
+
+#[test]
+fn test_clean_all_and_component() {
+    let temp_dir = setup_package("coverage_project");
+
+    let clean_components_state = CleanComponentsState {
+        coverage: false,
+        cache: true,
+        trace: true,
+        profile: false,
+    };
+    generate_clean_components(clean_components_state, &temp_dir);
+
+    // This command should fail because 'all' cannot be combined with other components
+    runner(&temp_dir)
+        .arg("clean")
+        .arg("all")
+        .arg("cache")
+        .assert()
+        .failure(); // "The 'all' component cannot be combined with other components"
```

Comment:
```suggestion
    // This command should fail because 'all' cannot be combined with other components
    runner(&temp_dir)
        .arg("clean")
        .arg("all")
        .arg("cache")
        .assert()
        .failure();
```